Userland/Utilities/acpi.cpp

```diff
@@ -0,0 +1,142 @@
+            out("{:#02x} ", data.at(pos + i));
+    else
+        out("               ");
+    out("  ");
+    for (auto i = 0; i < 5; i++)
+        out("{:#02x} ", data.at(pos + i));
+    outln("");
+
+    u32 length = 0;
+    u32 delta = 0;
+    const int size = data.at(pos + 1) >> 6;
+    if (size == 0) {
+        length = data.at(pos + 1) & 0x3f;
+        delta = 1;
+    } else if (size == 1) {
+        length = data.at(pos + 1) & 0x0f;
+        length = length | (data.at(pos + 2) << 4);
+        delta = 2;
+    } else if (size == 2) {
+        length = data.at(pos + 1) & 0x0f;
+        length = length | (data.at(pos + 2) << 4);
+        length = length | (data.at(pos + 3) << 12);
+        delta = 3;
+    } else {
+        length = data.at(pos + 1) & 0x0f;
+        length = length | (data.at(pos + 2) << 4);
+        length = length | (data.at(pos + 3) << 12);
+        length = length | (data.at(pos + 4) << 20);
+        delta = 4;
+    }
+    length -= (delta - 1);
+
+    outln("pos: {} {} {}", delta, pos + delta + length, data.size());
+    next_in_block = next_block + 1 + delta;
+    next_block = next_block + delta + length;
+}
+
+ErrorOr<int> serenity_main(Main::Arguments arguments)
+{
+    TRY(Core::System::pledge("stdio rpath"));
+
+    StringView filename;
+
+    Core::ArgsParser args_parser;
+    args_parser.add_positional_argument(filename, "Name of DSDT table", "DSDT table", Core::ArgsParser::Required::No);
+    args_parser.parse(arguments);
+
+    int fd;
+    if (filename.is_empty()) {
+        fd = STDIN_FILENO;
+    } else {
+        if (filename == "-") {
+            fd = 0;
+        } else {
+            auto fd_or_error = Core::System::open(filename, O_RDONLY);
+            if (fd_or_error.is_error()) {
+                warnln("Failed to open {}: {}", filename, fd_or_error.error());
+                return -1;
+            }
+            fd = fd_or_error.release_value();
+        }
+    }
+
+    TRY(Core::System::pledge("stdio"));
+
+    Array<u8, sizeof(SDTHeader)> header_buffer;
+    auto header_span = header_buffer.span();
+    auto nread = TRY(Core::System::read(fd, header_span));
+    if (nread != sizeof(SDTHeader)) {
+        outln("failed to read SDTHeader");
+    }
+
+    SDTHeader const* const header = reinterpret_cast<SDTHeader*>(header_buffer.data());
+
+    Vector<u8> data;
+    data.ensure_capacity(header->length);
+
+    Array<u8, 8192> buffer;
+    for (;;) {
+        auto buffer_span = buffer.span();
+        auto nread = TRY(Core::System::read(fd, buffer_span));
+        if (nread == 0) {
+            break;
+        }
+        buffer_span = buffer_span.trim(nread);
+        for (auto it = buffer_span.begin(); it != buffer_span.end(); it++) {
+            data.append(*it);
+        }
+    }
+
+    TRY(Core::System::close(fd));
+
+    u32 next_in_block = 0;
+    u32 next_block = 0;
+
+    while (next_block < data.size()) {
+        switch (data[next_block]) {
+        case 0x10: // ScopeOp
```

Comment:
Maybe prepare an enum for that,
if you feel fancy, even a simple stringifier

Reply:
I'll keep this for the next commit, where there is more than one Opcode

Reply:
Because that would lead down a rabbit-hole of learning...
I'm willing to invest the time but not for this first patch of a new application that has a big chance of being rejected

Reply:
Is there a particular reason why you can't just add an enum with a single value for now? I never said that it would have to be complete.

Reply:
There is no technical reason.
I just want a minimal viable implementation that does something interesting because the code is the easy part, making it through the first review and merge is hard.
By adding an enum this early I feel that we are leaving simplicity and entering more of a design first approach. That has been my work for many years and I want to learn other ways of working.